extensions/wikia/Email/EmailController.class.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Email;
+
+abstract class EmailController extends \WikiaController {
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	/** @var \User The user associated with the current request */
+	protected $currentUser;
+
+	/** @var \User The user this email will be sent to */
+	protected $targetUser;
+
+	protected $hasErrorResponse = false;
+
+	/** @var bool Whether or not to actually send an email */
+	protected $test;
+
+	/**
+	 * Since the children of this class are located in the 'Controller' directory, the default
+	 * location for the template directory would be 'Controller/templates'.  Redefine it to be
+	 * at the root of this extension instead.
+	 *
+	 * @return string
+	 */
+	public static function getTemplateDir() {
+		return dirname( __FILE__ ) . '/templates';
+	}
+
+	public function init() {
+		try {
+			$this->assertCanAccessController();
+
+			$this->currentUser = $this->findUserFromRequest( 'currentUser', $this->wg->User );
+			$this->targetUser = $this->findUserFromRequest( 'targetUser', $this->wg->User );
```

Comment:
At least in the case of forgot email, these can be (and are treated differently) different users.  There's the user who is making the request (likely a non-logged in anonymous user but could be a real user) and then there's the user from whatever 'username' is sent to us.

We want to check current user in the for being blocked below because that is the user who is actually logged in currently requesting to take an action.

So current user is whoever is taking the action, and target user is whoever the email is being sent to.  Most often they are the same.